### DISCUSSION TITLE
How to lazy load Architext

### DISCUSSION THREAD

**Discussion by AckslD** [wbthomason/packer.nvim#481] [Q&A]
I'm trying to lazy load [architext](https://github.com/vigoux/architext.nvim) with the following setup:
```lua
use { -- architext {{{
    "vigoux/architext.nvim",
    cmd = {'ArchitextREPL', 'Architext', 'A'},
} -- }}}
```
but if I type for example:
```
:Architext/((identifier) @i (#eq? @i "foo"))/i:bar/
```
from the README of architext I get the error:
```
E194: No alternate file name to substitute for '#'
```
If I try it again I get the same thing

But if I do just
```
:Architext
```
I first get the error:
```
Error in packer_compiled: ...vim/site/pack/packer/opt/packer.nvim/lua/packer/load.lua:77: Vim:E471: Argument required: Architext
Please check your config for correctness
```
But after that point I can use the command.

**Comment by wbthomason** [comment]
Ah, another interesting issue! I would expect what you have to work. Command lazy-loading passes any args via `<q-args>` to the `load` function, which then re-runs the command with those args. I'm guessing that using `<q-args>` is breaking something here, though I'm not entirely sure what - it's just that I notice that `architext` uses `<f-args>` in the commands it defines.

I think this question is probably actually an issue, so I'm going to convert it.

**Reply by AckslD** [reply]
Cool thanks @wbthomason
